src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -349,17 +350,26 @@ FancyZones::VirtualDesktopInitialize() noexcept
 IFACEMETHODIMP_(void)
 FancyZones::WindowCreated(HWND window) noexcept
 {
+    std::shared_lock readLock(m_lock);
     if (m_settings->GetSettings()->appLastZone_moveWindows && IsInterestingWindow(window))
     {
         for (const auto& [monitor, zoneWindow] : m_zoneWindowMap)
         {
+            GUID windowDesktopId{};

```

Comment:
Please add a comment explaining the context for this fix.
```
// WindowCreated is also invoked when a virtual desktop switch occurs, we need a way
// to figure out when that happens to avoid moving windows that should not be moved
```

Reply:
Comment added.